xbmc/cores/VideoPlayer/VideoPlayer.cpp

```diff
@@ -2439,6 +2440,16 @@ void CVideoPlayer::OnExit()
   if (!m_bAbortRequest)
     CLog::Log(LOGNOTICE, "VideoPlayer: eof, waiting for queues to empty");
 
+  VideoStreamInfo video;
+  AudioStreamInfo audio;
+  SubtitleStreamInfo subtitle;
+  if (m_item.GetProperty("get_stream_details_from_player").asBoolean())
```

Comment:
You could make VP update VideoInfoTag with current streams by explicitly calling a method from outside, something like UpdateFilemWithCurrentSteam. That is better than moving specifics like get_stream_details_from_player into player.

Reply:
And please keep the changes into a single block

```
if (m_UpdateVideoInfoTag)
{
  all changes here
}
```